### PR TITLE
program: add switchboard

### PR DESCRIPTION
**Pull request by lilperp**

### CODE REVIEW

File: programs/drift-competitions/src/instructions/request_randomness.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    )?;
+
+    Ok(())
+}
+
+#[derive(Accounts)]
+pub struct RequestRandomness<'info> {
+    // COMPETITION ACCOUNTS
+    #[account(mut)]
+    pub sponsor: Signer<'info>,
```
Comment (0xbigz) [inline programs/drift-competitions/src/instructions/request_randomness.rs]: so only the sponsor can request randomness?

Comment (lilperp) [reply programs/drift-competitions/src/instructions/request_randomness.rs]: nah i changed it

File: programs/drift-competitions/src/instructions/receive_randomness.rs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+use switchboard_solana::prelude::*;
+use crate::state::Competition;
+
+pub fn receive_randomness(_ctx: Context<ReceiveRandomness>, winner_randomness: u128, prize_randomness: u128) -> Result<()> {
+    msg!("winner_randomness {}", winner_randomness);
+    msg!("prize_randomness {}", prize_randomness);
+
+    let mut competition = _ctx.accounts.competition.load_mut()?;
+    competition.winner_randomness = winner_randomness;
+    competition.prize_randomness = prize_randomness;
```
Comment (0xbigz) [inline programs/drift-competitions/src/instructions/receive_randomness.rs:11]: i named these `_draw` in the competition struct, should that just be set here and remove the `_randomness`?

File: programs/drift-competitions/src/instructions/update_switchboard_function.rs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+use anchor_spl::associated_token::AssociatedToken;
+use anchor_spl::token::{Mint, Token};
+use switchboard_solana::prelude::*;
+use crate::signer_seeds::get_function_authority_seeds;
+use crate::state::Competition;
+
+/// The minimum guess that can be submitted, inclusive.
+pub const MIN_RESULT: u32 = 1;
+/// The maximum guess that can be submitted, inclusive.
+pub const MAX_RESULT: u32 = 1_000_000;
```
Comment (0xbigz) [inline programs/drift-competitions/src/instructions/update_switchboard_function.rs]: is this just an example or hardcoded upper bound? this bound should be much higher if it even needs to exist at all

File: switchboard-function/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        let max = 9;
+
+        let mut counts = vec![0; 10];
+        for _ in 0..1000 {
+            let result = generate_randomness(min, max);
+            let index: usize = result as usize;
+            counts[index] += 1;
+        }
+
+        // Ensure all counts are non-zero (probabilistically should be the case)
```
Comment (0xbigz) [inline switchboard-function/src/main.rs:127]: why not just keep pulling until they all have one (to avoid random failures)

Comment (lilperp) [reply switchboard-function/src/main.rs:127]: i just copied this lol

**Pull request merged**